kernel/arch/dreamcast/include/dc/biosfont.h

```diff
@@ -415,14 +470,56 @@ unsigned char bfont_draw_wide(void *buffer, uint32 bufwidth, uint8 opaque,
                             otherwise do not change them from what they are.
     \param str              The string to draw.
 
+    \sa bfont_draw_str_ex_fmt(), bfont_draw_str_ex_va()
+*/
+void bfont_draw_str_ex(void *b, uint32_t width, uint32_t fg, uint32_t bg,
+                       uint8_t bpp, bool opaque, const char *str);
+
+/** \brief   Draw a full decorated and formatted string to any sort of buffer.
+
+    This function is equivalent to bfont_draw_str_ex(), except that the string
+    is formatted as with the `printf()` function.
+
+    \param b                The buffer to draw to.
+    \param width            The width of the buffer in pixels.
+    \param fg               The foreground color to use.
+    \param bg               The background color to use.
+    \param bpp              The number of bits per pixel in the buffer.
+    \param opaque           If non-zero, overwrite background areas with black,
+                            otherwise do not change them from what they are.
+    \param fmt              The printf-style format string to draw.
+    \param ...              Additional printf-style variadic arguments
+
+    \sa bfont_draw_str_ex_vfmt()
 */
-void bfont_draw_str_ex(void *b, uint32 width, uint32 fg, uint32 bg, uint8 bpp,
-                       uint8 opaque, const char *str);
+void bfont_draw_str_ex_fmt(void *b, uint32_t width, uint32_t fg, uint32_t bg,
+                           uint8_t bpp, bool opaque, const char *fmt, ...)
+                           __printflike(7, 8);
+
+/** \brief   Draw decorated and formatted string to buffer (with va_args).
+
+    This function is equivalent to bfont_draw_str_ex_fmt(), except that the
+    variadic argument list is passed via a pointer to a va_list.
+
+    \param b                The buffer to draw to.
+    \param width            The width of the buffer in pixels.
+    \param fg               The foreground color to use.
+    \param bg               The background color to use.
+    \param bpp              The number of bits per pixel in the buffer.
+    \param opaque           If non-zero, overwrite background areas with black,
+                            otherwise do not change them from what they are.
+    \param fmt              The printf-style format string to draw.
+    \param var_args         Additional printf-style variadic arguments
+
+    \sa bfont_draw_str_ex_fmt()
+*/
+void bfont_draw_str_ex_vfmt(void *b, uint32_t width, uint32_t fg, uint32_t bg,
+                            uint8_t bpp, bool opaque, const char *fmt,
+                            va_list *var_args);
 
 /** \brief   Draw a full string to a buffer.
-    \ingroup bfont
 
-    This function draws a NUL-terminated string in the set encoding to the given
```

Comment:
It may be that it's more confusing than helpful, but `NUL` was purposeful in this context for the name of the character that is at encoding 0 (null) in string encoding schemes. To differentiate from NULL as an invalid pointer (even though of course they all have the same value and whatnot).